### ISSUE TITLE
Add DOM types to ReactTags?

### ISSUE THREAD

**Issue by japgolly** [japgolly/scalajs-react#199]
@elacin said:

That same commit (https://github.com/elacin/scalajs-react/commit/fb6d0323b655ae80c3f005e161b3d164ff90544f) also shows an experiment with typing `ReactTag`s (so i can haz some type-safety for my `HTMLInputElement => Unit` ref updates), would that be something you're interested in?

About typed `ReactTag`s, I can probably save you the trouble. it comes down to this:
```scala
//covariant, to let a ReactTagOf[html.Paragraph] also be a ReactTagOf[TopNode]
abstract class ReactTagOf[+N <: TopNode] extends DomFrag 
//regain invariance to make compiler happy
private [vdom] final case class ReactTagImpl[N <: TopNode] 
//make sure existing code should be allright
final type ReactTag = japgolly.scalajs.react.vdom.ReactTagOf[TopNode]

// make all tags in HtmlTags.scala gain more specified types
implicit class STStringExt(private val s: String) extends AnyVal {
  def tag[N <: TopNode](implicit namespaceConfig: Namespace): ReactTagOf[N] = ...
  ...
}
```

**Comment by japgolly** [comment]
I think this would be a great idea but it's not high on the list of things I would implement myself. I was thinking about doing this a while back but guessed that it would change too many existing types and decided against it. I quite liked the idea of the component's DOM type being automatically inferred by the result of the render function but then thought that would be too disruptive.

If you delivered it in one of your awesome PRs it would be very welcome! So long as it doesn't add much burden to users/callsites. :)

**Comment by elacin** [comment]
Cool. I'll see if i can find the time for that tonight :)

**Comment by japgolly** [comment]
Closed via #202
